lib/components/src/Accordion/AccordionItem.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useCallback, useContext, useEffect, useRef, useState, Children } from 'react';
+import { uniqueId } from 'lodash';
+import { styled } from '@storybook/theming';
+import { AccordionContext } from './AccordionContext';
+import { AccordionItemContext } from './AccordionItemContext';
+
+// Props are also available from Accordion context provider, but local props
+// takes predecense for scope control
+export type AccordionItemProps = {
+  open?: boolean;
+  indentBody?: boolean;
+  lined?: boolean;
+  narrow?: boolean;
+  preventToggle?: boolean;
+} & React.HTMLAttributes<HTMLLIElement>;
+
+export const AccordionItem = ({
+  children,
+  open: _open,
+  indentBody: _indentBody,
+  narrow: _narrow,
+  lined: _lined,
+  preventToggle: _preventToggle,
+  ...rest
+}: AccordionItemProps) => {
+  const [open, setOpen] = useState(_open);
+  const context = useContext(AccordionContext);
+  const id = useRef(uniqueId('AccordionItem-'));
+  const preventOpen = Children.count(children) < 2;
+  const initialOpen = useRef(_open);
+
+  const onExpand = useCallback(() => {
+    if (!preventOpen && _open !== true) {
```

Comment:
what about assigning this to a variable with a meaningful name? this is used in multiple places